index/scorch/segment/zap/docvalues.go

```diff
@@ -178,3 +179,22 @@ func (s *Segment) VisitDocumentFieldTerms(localDocNum uint64, fields []string,
 	}
 	return nil
 }
+
+// VisitableDocValueFields returns the list of fields with
+// persisted doc value terms ready to be visitable using the
+// VisitDocumentFieldTerms method.
+func (s *Segment) VisitableDocValueFields() ([]string, error) {
+	if len(s.fieldsInv) == 0 {
```

Comment:
i think we can just remove this if check, if we leave it out, we return nil, nil anyway and i don't think there is any significant gain by doing it explicitly.